paypal/donations-pp.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+""" Create a CSV to import into CiviCRM from PayPal output CSV.
+
+Fields order or their availability are not an issue with this script. But *this*
+*does not mean resulting csv file will import into CiviCRM successfully*.
+
+To use this python script run:
+~~~bash
+python export.py ${paypal_csv} ${export_folder}
+~~~
+
+#Other info
+- the date format seems to be %d/%m/%Y
+- header line is needed for the importing PayPal csv file
+"""
+
+import os
+import csv
+import sys
+import datetime
+
+# Fields to export==============================================================
+# Variable names are the name that the fields should import into
+
+# The values here are the files in the PayPal csv file
+
+PAYOUT_DATE="Payout Date"
+DONATION_DATE="Donation Date"
+FIRST_NAME="Donor First Name" #REQUIRED
+LAST_NAME="Donor Last Name" #REQUIRED
+EMAIL="Donor Email" #REQUIRED
+PROGRAM_NAME="Program Name"
+REFERENCE="Reference Information"
+CURRENCY="Currency Code"
+GROSS_AMOUNT="Gross Amount" #REQUIRED
+TOTAL_FEES="Total Fees"
+NET_AMOUNT="Net Amount"
+TRANSACTION_ID="Transaction ID"
+
+EXTERNAL_ID="Donor Email" #REQUIRED
+
+# Values used the export value
+FINANCIAL_TYPE = "Donation" #REQUIRED
+PAYMENT_METHOD = "PayPal"
+MEMBERSHIP_TYPE = "VCN Member"
+
+# Constants used in this file===================================================
+IND_CONTACT_FILE = "/individual_contacts.csv"
+IND_DONATION_FILE = "/individual_donations.csv"
+MEMBERSHIP_FILE = "/memberships.csv"
+
+MEMBERSHIP_MIN_AMOUNT = 15
+
+INPUT_DATE_FORMATS = ['%d/%m/%Y']
+OUTPUT_DATE_FORMAT = '%Y-%m-%d'
+
+# Main export function==========================================================
+
+def export(fileName, outputFolder):
+	""" Export the four files
+
+	Arugment:
+		fileName     -- (String) the input csv file path
+		outputFolder -- (String) the output folder path
+	"""
+	if not(os.path.exists(outputFolder)):
+		os.makedirs(outputFolder)
+
+	# output list
+	ind_contacts = []
+	ind_donations = []
+	memberships = []
+
+	# open the export file as a text file of given encoding
+	# (and implicitly convert to utf-8 in python)
+	with open(fileName, mode='r', encoding='utf-8-sig') as ppFile:
```

Comment:
This isn't work. The files are ASCII format.

Reply:
Ah, didn't know that vanilla ASCII is a valid encoding--figuring out encoding is hard sometimes